src/s1reader/s1_info.py

```diff
@@ -48,27 +56,109 @@ def _plot_bursts(safe_path, output_dir="burst_maps"):
+    use the union of all burst bounds (which is slower).
+
+    Parameters
+    ----------
+    safe_path : Union[Path, str]
+        Path to the SAFE directory or zip file.
+
+    Returns
+    -------
+    List[float]
+        [lonmin, latmin, lonmax, latmax]
+    """
+    try:
+        return _bounds_from_preview(safe_path)
+    except Exception as e:
+        warnings.warn(f"Could not get bounds for {safe_path}: {e}")
+    return _bounds_from_bursts(safe_path)
+
+
+def _bounds_from_preview(safe_path: Union[Path, str]) -> List[float]:
+    """Get the bounding box of the frame from the preview/map-overlay.kml."""
+    # looking for:
+    # S1A_IW_SLC__1SDV_20221005T125539_20221005T125606_045307_056AA5_CB45.SAFE/preview/map-overlay.kml
+    if _is_safe_dir(safe_path):
+        overlay_path = Path(safe_path) / "preview" / "map-overlay.kml"
+        temp_dir = None
+    else:
+        # The name of the unzipped .SAFE directory (with .zip stripped)
+        temp_dir = tempfile.mkdtemp()
+        with zipfile.ZipFile(safe_path, "r") as zip_ref:
+            zname = [
+                zi
+                for zi in zip_ref.infolist()
+                if "preview/map-overlay.kml" in zi.filename
+            ]
+            overlay_path = Path(zip_ref.extract(zname[0], path=temp_dir))
+
+    # Check that they have all the necessary kmls
+    if not overlay_path.exists():
+        raise ValueError(f"{overlay_path} does not exist.")
+    root = ET.parse(overlay_path).getroot()
+
+    # point_str looks like:
+    # <coordinates>-102.552971,31.482372 -105.191353,31.887299...
+    point_str = list(elem.text for elem in root.iter("coordinates"))[0]
+    coords = [p.split(",") for p in point_str.split()]
+    lons, lats = zip(*[(float(lon), float(lat)) for lon, lat in coords])
+    if temp_dir is not None:
+        shutil.rmtree(temp_dir)
+    return [min(lons), min(lats), max(lons), max(lats)]
+
+
+def _bounds_from_bursts(safe_path: Union[Path, str]) -> List[float]:
+    """Get the bounding box of the frame from the union of all burst bounds."""
+    # Get all the bursts from subswath 1, 2, 3
+    bursts = get_bursts(safe_path)
+    # Convert the border (list of polygons) into a MultiPolygon
+    all_borders = [shapely.geometry.MultiPolygon(b.border) for b in bursts]
+    # Perform a union to get one shape for the entire frame
+    border_geom = shapely.ops.unary_union(all_borders)
+    # grab the bounds and pad as needed
+    return list(border_geom.bounds)
+
+
 EXAMPLE = """
 Example usage:
 
     # Print all bursts in a Sentinel-1 SLC product
-    s1_info.py S1A_IW_SLC__1SDV_20180601T000000_20180601T000025_021873_025F3D_9E9E.zip
+    s1_info S1A_IW_SLC__1SDV_20180601T000000_20180601T000025_021873_025F3D_9E9E.zip
 
     # Print only the burst IDs
-    s1_info.py S1A_IW_SLC__1SDV_20180601T000000_20180601T000025_021873_025F3D_9E9E.SAFE --burst-id
+    s1_info S1A_IW_SLC__1SDV_20180601T000000_20180601T000025_021873_025F3D_9E9E.SAFE --burst-id
```

Comment:
I would suggest to add bounding box printing for each burst as well. Maybe an add on here to show both burst ids and also bounding box

Reply:
I added the option `s1_info S1A_IW_SLC__1SDV_20180601T000000_20180601T000025_021873_025F3D_9E9E.SAFE --burst-bbox` (or which you can add to the other ones like `s1_info S1A_IW_SLC__1SDV_20180601T000000_20180601T000025_021873_025F3D_9E9E.SAFE -b --burst-bbox` to only print the burst ID, but also the bounding box)

I changed the other one to `--frame-bbox` for clarity.

```
(mapping) staniewi:s1-reader$ s1_info  S1A_IW_SLC__1SDV_20220226T124745_20220226T124812_042084_050378_F69A.SAFE/ --burst-bbox -b
Found 1 Sentinel-1 SLC products.
Bursts in S1A_IW_SLC__1SDV_20220226T124745_20220226T124812_042084_050378_F69A.SAFE:
--------------------------------------------------------------------------------
t012_024518_iw1 [-101.5785461794019, 30.88066805649816, -100.5878652599923, 31.20102438050872]
t012_024519_iw1 [-101.6137935302681, 30.71434634420109, -100.6261288577148, 31.03472849442007]
t012_024520_iw1 [-101.648984853532, 30.5481232100879, -100.6631136010606, 30.86845468759752]
...
```